pueue/src/daemon/service.rs

```diff
@@ -0,0 +1,640 @@
+use std::{
+    env,
+    ffi::{c_void, OsString},
+    iter,
+    path::PathBuf,
+    process, ptr,
+    sync::{
+        atomic::{AtomicBool, Ordering},
+        mpsc::{channel, Receiver, Sender},
+        Arc, Mutex,
+    },
+    thread,
+    time::Duration,
+};
+
+use anyhow::{anyhow, bail, Result};
+use log::{debug, error};
+use once_cell::sync::OnceCell;
+use windows::{
+    core::{PCWSTR, PWSTR},
+    Win32::{
+        Foundation::{CloseHandle, HANDLE, LUID},
+        Security::{
+            AdjustTokenPrivileges, DuplicateTokenEx, LookupPrivilegeValueW, SecurityIdentification,
+            TokenPrimary, SE_PRIVILEGE_ENABLED, SE_PRIVILEGE_REMOVED, SE_TCB_NAME,
+            TOKEN_ACCESS_MASK, TOKEN_ADJUST_PRIVILEGES, TOKEN_PRIVILEGES,
+        },
+        System::{
+            Environment::{CreateEnvironmentBlock, DestroyEnvironmentBlock},
+            RemoteDesktop::{WTSGetActiveConsoleSessionId, WTSQueryUserToken},
+            SystemServices::MAXIMUM_ALLOWED,
+            Threading::{
+                CreateProcessAsUserW, GetExitCodeProcess, OpenProcess, OpenProcessToken,
+                TerminateProcess, WaitForSingleObject, CREATE_NO_WINDOW,
+                CREATE_UNICODE_ENVIRONMENT, INFINITE, PROCESS_INFORMATION,
+                PROCESS_QUERY_INFORMATION, STARTUPINFOW,
+            },
+        },
+    },
+};
+use windows_service::{
+    define_windows_service,
+    service::{
+        ServiceAccess, ServiceControl, ServiceControlAccept, ServiceErrorControl, ServiceExitCode,
+        ServiceInfo, ServiceStartType, ServiceState, ServiceStatus, ServiceType,
+        SessionChangeParam, SessionChangeReason, SessionNotification,
+    },
+    service_control_handler::{self, ServiceControlHandlerResult},
+    service_dispatcher,
+    service_manager::{ServiceManager, ServiceManagerAccess},
+};
+
+#[derive(Clone)]
+struct Config {
+    config_path: Option<PathBuf>,
+    profile: Option<String>,
+}
+
+const SERVICE_NAME: &str = "pueued";
+const SERVICE_TYPE: ServiceType = ServiceType::OWN_PROCESS;
+static CONFIG: OnceCell<Config> = OnceCell::new();
```

Comment:
Not sure once_cell is needed anymore as a dependency?

https://www.reddit.com/r/rust/comments/126qaai/after_years_of_work_and_discussion_once_cell_has/

Reply:
You'll need to [raise your msrv to 1.80](https://github.com/Nukesor/pueue/blob/1c5f4aa06d1ffa9081d3eea40b77d0e5d0d137cd/Cargo.toml#L14) before we can use `LazyLock`

Reply:
@Nukesor Do you think it would be a problem to do that, I don't see an MSRV policy in the Readme, so not sure if there is a policy on this?

Reply:
Not really. As far as I'm aware, packagers usually roll their own setups and don't necessarily depend on the currently packaged rust version for a given OS. So depending on `stable` shouldn't be a problem :)

Feel free to bump the MSRV